speechcolab/datasets/gigaspeech.py

```diff
@@ -1,15 +1,47 @@
 from pathlib import Path
 
-allowed_subsets = ('XS', 'S', 'M', 'L', 'XL')
+import ijson
 
 
 class GigaSpeech(object):
     def __init__(self, path='.'):
-        self.path = path
+        self.root_path = Path(path)
+        self.json_path = self.root_path / 'GigaSpeech.json'
```

Comment:
I’m curious if it’s not preferable to store large manifests like these in JSONL (JSON per line) instead? You could avoid the extra dependency (although I don’t know how your JSON is structured yet so maybe what I’m suggesting is not viable). Anyway that format seems to work well in Lhotse and I’m currently adding more support for handling large datasets without memory/reading time overhead - if you’re going to use Lhotse for training on GigaSpeech, don’t hesitate to ask me about it, I am going to document that in the near future.

Reply:
Here is a snippet of the JSON format metadata:

```json
{
    "dataset": "GigaSpeech",
    "description": "GigaSpeech is a large audio dataset which contains 10k hours of annotated English speech, covering audio book, podcast and youtube videos",
    "language": "EN",
    "version": "v0.2.5",
    "audios": [
        {
            "title": "Check Cashing Stores",
            "url": "https://99percentinvisible.org/episode/episode-18-check-cashing-stores-download-embed/download",
            "path": "audio/podcast/P0001/POD0000000001.opus",
            "aid": "POD0000000001",
            "source": "podcast",
            "format": "opus",
            "sample_rate": 16000,
            "codec": "s16le",
            "channels": "1",
            "license": "TBD",
            "md5": "a1eb19f5bc68d4af71f65a3e8d700343",
            "duration": "629.9308",
            "size": "2555888",
            "speaker": "N/A",
            "category": "News",
            "subsets": [
                "{XL}",
                "{L}"
            ],
            "segments": [
                {
                    "sid": "POD0000000001_S0000008",
                    "speaker": "N/A",
                    "begin_time": 159.0,
                    "end_time": 167.52,
                    "text_raw": "",
                    "text_tn": "DOUGLAS MCGRAY IS GOING TO BE OUR GUIDE YOU WALK THROUGH THE DOOR <COMMA> YOU SEE THE RED CARPETING <COMMA> YOU SEE SOMEONE IN A SUIT <PERIOD> THEY MAY BE GREETING YOU <PERIOD>",
                    "subsets": [
                        "{XL}"
                    ]
                },
                {
                    "sid": "POD0000000001_S0000010",
                    "speaker": "N/A",
                    "begin_time": 195.46,
                    "end_time": 208.22,
                    "text_raw": "",
                    "text_tn": "AH ITALIAN <QUESTIONMARK> AND AND THEN YOU HAVE A CONVERSATION ABOUT WHAT THEY COULD PROVIDE AND AND THE SEVENTEEN DIFFERENT SAUCES THEY COULD OFFER AND ALL THE DIFFERENT SHAPES OF PASTA <PERIOD> A REASONABLE PERSON MIGHT THINK LIKE IT WOULD BE KIND OF NICE TO HAVE A MENU",
                    "subsets": [
                        "{XL}"
                    ]
                },
``` 

The complexity is that we have two levels: GigaSpeech --> audios --> segments. But yes I think it's a good idea to provide the corresponding JSONL file as well.

One option is that we provide a script to convert the JSON format metadata to JSONL files, perhaps one for audio information, and another for segment information, and the segment file will be something like:

```jsonl
{"sid": ..., "speaker": ..., "begin_time": ..., "end_time": ..., "text_raw": ..., "text_tn": ..., "subsets": ..., audio: ...}
{...}
```

What are your thoughts? We are open to suggestions. @pzelasko